### DISCUSSION TITLE
New Beta v1.3.2.4 Software Release with Command and Control

### DISCUSSION THREAD

**Discussion by alanbjohnston** [alanbjohnston/CubeSatSim#335] [Announcements]
Hi everyone,

There is a new beta software release v1.3.2.4 available for the CubeSatSim.  You can see the full details here:

https://github.com/alanbjohnston/CubeSatSim/releases/tag/v1.3.2.4

You can update using the git commands, or you could flash a new SD card using this image: 

http://cubesatsim.org/download/cubesatsim-v1.3.2.4.iso.zip

The main additional feature is Command and Control using DTMF tones or APRS packets.  This can be great for demos to use your UHF HT to change telemetry modes or turn the transmitter on and off using radio signals.

To get all the features, there is some new hardware required.  The BOM has been updated with the new parts and the Wiki instructions, too.  If you have an already built v1.3.2 CubeSatSim (blue PCBs) or are building one right now, it is easy to add the components.

Full details are here [on the Command and Control wiki page](https://github.com/alanbjohnston/CubeSatSim/wiki/Command-and-Control).

Feel free to post any questions or feedback on these new capabilities.

Alan

**Comment by F4LBJ** [comment]
Hello Alan,

Can't wait to test this new version.
However, I was unable to install it from the image file. The file is too big 15.9GB! compared to 3.97GB for version 1.3.2.3. My SD card won't accept more than 15.6GB.  Wasn't there an error when creating the ISO Image?

Best regards,

Arnaud
73 !

**Reply by alanbjohnston** [reply]
Hi Arnaud,

Sorry to hear it didn't work.  I just tested it and it worked for me.  Here's what I did.

I used this link: http://cubesatsim.org/download/cubesatsim-v1.3.2.4.iso.zip which is a 2.2G file

I used balenaEtcher on a Mac with Flash from URL option, and a 16GB SD card.  My SD card does show a size of 15.9GB, so this might be the issue.

What process and software did you try?  

I also tried downloading the file, which resulted in having the file on my computer with a size of 2,360,614,209 bytes .  When expanded, it was 15,931,539,456 bytes.

If you have an earlier v1.3.2 image, such as v1.3.2.3, you can try updating it by logging into your Pi Zero and typing:

```
cd

CubeSatSim/update
```

You might need to force a reboot by typing:

`sudo reboot now`

Let me know if this works for you.

73,
Alan

**Reply by F4LBJ** [reply]
Hi Alan,
Thanks for your reply. I use the same method as you to program the SD card (with balenaEtcher) so I think my SD cards are too limited in space.
I was able to upgrade with the command: CubeSatSim/update. I have the new functions but I don't know if everything went well because I still have the problem of the Cubesat cutting off (after 5 flashes of the green LED and then slow flashes). The Cubesat restarts but does not stay cut off.
Another question, I tested the transponder mode, when I transmit to the Cubesat, the squelch is open on the reception frequency (434.9) but my voice is not transmitted. Do I need to add the sound card?

Finally, I hadn't kept you informed, but I gave a presentation on the CubeSatSim at the AMSAT Francophone meetings. You can find the video here: https://www.youtube.com/watch?v=0Uz0eR22Iiw
The audience showed a strong interest in the project.
The CubeSat was also presented by AMSAT Francophone on the AMSAT DL stand at the Friedrichshafen trade fair, you can find the photos here: https://site.amsat-f.org/2024/06/28/friedrichshafen-lamsat-f-vous-rencontre-sur-le-stand-de-lamsat-dl/
![image](https://github.com/user-attachments/assets/7a7c1820-c909-4cf8-b0b4-5e765e129142)

73 !
Arnaud

**Reply by alanbjohnston** [reply]
Hi Arnaud,

Thanks for sharing the link and the photos!

I will try generating a new image to see if it is smaller.

Transponder mode does require the USB sound card in order to work - that is the only way the Pi Zero is able to receive the audio.

You can try to manually fix the push button shutdown bug by running these commands:

       cd /home/pi/pi-power-button

       git pull

       git checkout reboot-mode-change-beta
    
       script/install

That should fix it.

The other thing you can try is doing a fresh install of the CubeSatSim software starting from a fresh image of `Raspberry Pi OS (Legacy, 32-bit) Lite.`  You can follow the steps here which show how to use the Raspberry Pi Imager application:

https://github.com/alanbjohnston/CubeSatSim/wiki/V2-Pi-Zero-Software

Let me know if any of these approaches work.

73,
Alan

**Reply by F4LBJ** [reply]
Hi Alan,
All right, it's working!
I used the commands you gave to upgrade the pi-power-button package, and now the CubeSat cuts off correctly.
Thanks a lot.
 I'll now look at adding the sound card and testing the other functions.
If you have the possibility of compiling a disk image a little lighter I am interested.

73 !
Arnaud
